src/figuresClasses.ts

```diff
@@ -1,11 +1,102 @@
-export interface Figure {}
+export enum Shape {
+  Triangle = 'triangle',
+  Rectangle = 'rectangle',
+  Circle = 'circle',
+}
+
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public readonly shape: Shape = Shape.Triangle;
 
-export class Triangle implements Figure {}
+  constructor(
+    public readonly color: Color,
+    public readonly a: number,
+    public readonly b: number,
+    public readonly c: number,
+  ) {
+    if ([a, b, c].some((side) => side <= 0)) {
+      throw new Error('All sides must be > 0');
+    }
+
+    const longestSide = Math.max(a, b, c);
+    const otherSides = [a, b, c].filter((side) => side !== longestSide);
+
+    if (longestSide >= otherSides[0] + otherSides[1]) {
+      throw new Error(
+        `Triangle with sides [ a = ${a}, b = ${b}, c = ${c} ] is impossible`,
+      );
```

Comment:
[CODE STYLE] - The error message thrown here is good as it provides a detailed description of the issue.